### PR TITLE
Fixed issue [security] #19133: User can send email to group members, while not having permissions

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: only $userGroup->owner_id (and superadmin) can send email Usergroup

### CODE REVIEW

File: application/controllers/UserGroupController.php [hunk 1]
```diff
@@ -467,37 +460,36 @@ public function actionMailToAllUsersInGroup(int $ugid)
...
-                    $this->redirect(array('userGroup/viewGroup/ugid/' . $ugid));
-                }
-            } else {
-                Yii::app()->user->setFlash('error', gT("You do not have permission to send emails to all users. "));
-                $this->redirect(array('userGroup/viewGroup/ugid/' . $ugid));
-            }
-        } else {
-            $aData['ugid'] = $ugid;
+            $this->redirect($redirectUrl);
+            App()->end(); // redirect end : add it here for clarity
```
Comment (gabrieljenik) [inline application/controllers/UserGroupController.php:492]: `App()->end()` or `return` ?

File: application/controllers/UserGroupController.php [hunk 2]
```diff
@@ -467,37 +460,36 @@ public function actionMailToAllUsersInGroup(int $ugid)
     {
         $ugid = sanitize_int($ugid);
         $action = Yii::app()->request->getPost("action");
+        $currentUserId = App()->getCurrentUserId();
+        $userGroup = UserGroup::model()->findByPk($ugid);
+        if (empty($userGroup)) {
+            throw new CHttpException(404, gT("User group not found."));
+        }
+        if (
+            !Permission::model()->hasGlobalPermission('superadmin', 'read') // User is superadmin
```
Comment (gabrieljenik) [inline application/controllers/UserGroupController.php:469]: I would change the comments: user is not superadmin and is not owner.

File: application/controllers/UserGroupController.php [hunk 3]
```diff
@@ -158,8 +150,6 @@ public function actionViewGroup($ugid, bool $header = false)
             $aNewUserListData = CHtml::listData($aUsers, 'uid', function ($user) {
                 return \CHtml::encode($user->users_name) . " (" . \CHtml::encode($user->full_name) . ')';
             });
-            // Remove group owner because an owner is automatically member of a group
```
Comment (gabrieljenik) [inline application/controllers/UserGroupController.php:161]: The comment is being removed. Doesnt apply anymore?
If so, the unset below, should til be there?

**Review by gabrieljenik — APPROVED** [review]
Code structure looks OK. A few comments to review but could be nothing

File: application/controllers/UserGroupController.php (hunk 1 above)
Comment (Shnoulle) [reply application/controllers/UserGroupController.php:492]: Redirect already App->end, advice take from a comment of @mfavetti

File: application/controllers/UserGroupController.php (hunk 3 above)
Comment (Shnoulle) [reply application/controllers/UserGroupController.php:161]: You're right … not related to this commit.

OIn 6X : we can remove the owner form user in group, not in 5X
Unsure we need to remove it

**Comment by Shnoulle** [comment]
5X version https://github.com/LimeSurvey/LimeSurvey/pull/3586

**Pull request merged**